pages/developers/nox.md

```diff
@@ -0,0 +1,255 @@
+---
+layout: page
+title: Task runners
+permalink: /developer/tasks
+nav_order: 15
+parent: Developer information
+---
+
+# Task runners
+
+A task runner, like [make][] (fully general), [rake][] (Ruby general),
+[invoke][] (Python general), [tox][] (Python packages), or [nox][] (Python
+simi-general), is a tool that lets you specify a set of tasks via a common
+interface. These have been discouraged in Scikit-HEP in the past, since they
+can be a crutch, allowing poor packaging practices to be employed behind a
+custom script, and they can hide what is actually happening.
+
+We are carefully allowing an exception: [nox][]. Nox has two strong points that
+help with the above concerns. First, it is very explicit, and even prints what
+it is doing as it operates. Unlike the older tox, it does not have any implicit
+assumptions built-in. Second, it has very elegant build-in support for both
+virtual and Conda environments. This can greatly reduce new contributor
+friction with your codebase.
+
+A daily developer is not expected to use nox for simple tasks, like running
+tests or linting. You should _not_ rely on nox to make a task that should be
+made simple and standard (like building a package) complicated. You are not
+expected to use nox for linting on CI, or often even for testing on CI, even if
+those tasks are provided for users. Nox is a few seconds slower than running
+directly in a custom environment - but for new users, and rarely run tasks, it
+is _much_ faster than explaining how to get setup or manually messing with
+virtual environments.
+
+You _should_ use nox to make it easy and simple for new contributors to run
+things. You _should_ use nox to make specialized developer tasks easy. You
+_should_ use nox to avoid making single-use virtual environments for docs and
+other rarely run tasks.
+
+Nox doesn't handle binary builds very well, so for compiled projects, it might be
+best left to just specialized tasks.
+
+
+[nox]: https://nox.thea.codes
+[tox]: https://tox.readthedocs.io
+[invoke]: https://www.pyinvoke.org
+[rake]: https://ruby.github.io/rake/
+[make]: https://www.gnu.org/software/make/
+
+## Nox
+
+### Installing
+
+Installing nox should be handled like any other Python _application_. You should
+either use a good package manager, like brew on macOS, or you should use pipx; either
+permanently (`pipx install nox`) or by running `pipx run nox` instead of `nox`.
+
+
+On GitHub Actions or Azure, pipx is available by default, so you should use
+`pipx run nox`. To give it access to all Python versions, you can use this action:
```

Comment:
This feels a bit at odds with the above 

> You are not expected to use nox for linting on CI, or often even for testing on CI, even if those tasks are provided for users.

Is this just to show how one _would_ use `nox` with GHA? If so, do we need to show this if we're advocating for writing explicit CI? Or is there a use case here for `nox` in CI that I'm missing?

Reply:
This is to show how one would use it, yes. If you _just_ have simple linting and tests, that's likely better run explicitly without nox. However, if you do need it, this is how you'd do it. See https://github.com/scikit-hep/cookie/pull/19